src/framework/preferences/js/StarterSchemas.js

```diff
@@ -21,11 +21,41 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
      * contrast, table of contents, inputs larger and emphasize links
      *******************************************************************************/
 
+    var check;
+    var widthCheck = function() {
+        var width = $(window).width();
+        check = (width<640);
+        if (check) {
+            fluid.contextAware.makeChecks({
+                "fluid.responsiveCheck": true
+            });
+        }
+        else {
+            fluid.contextAware.forgetChecks("fluid.responsiveCheck");
+            fluid.defaults("fluid.prefs.auxSchema.starter.default", {
+                gradeNames: ["fluid.contextAware"],
+                contextAwareness: {
+                    sliderVariety: {
+                        checks: {
+                            jQueryUI: {
+                                contextValue: "{fluid.prefsWidgetType}",
+                                equals: "jQueryUI",
+                                gradeNames: ["fluid.prefs.auxSchema.starter.textSize.jQueryUI", "fluid.prefs.auxSchema.starter.lineSpace.jQueryUI", "fluid.prefs.auxSchema.starter.blueColorFilter.jQueryUI"]
```

Comment:
I tried to do that, but the problem is that the we must have the responsive check first and if it fails then we have to check for jQuery or native slide. In each case we had a different template to be applied and they did not override each other. There was also a problem with nesting contextAwarenesses. I tried I lot of different versions and this is the only one that worked.